docs/user/server-side-search/syntax.rst

```diff
@@ -59,6 +59,31 @@ user
 
    - ``user:@me test``
 
+Aliases
+~~~~~~~
+
+Some parameters support aliases as their values,
+aliases start with `@`.
+
+Depending on the context of the query, some aliases are available:
+
+@me
+   The current user.
+   This is only available when using the search API while authenticated.
+
+   Examples:
+
+   -  ``user:@me test``
+
+@this
+   The project and version of the current page.
+   This is only available when the current version and project are included in the API call.
```

Comment:
This feels really confusing to the user. How are they meant to know when this will work? I don't know if this is the best solution to this problem.

Can you clarify a bit more the use case you're imagining this being used on? If it's in-doc search, it seems that we already have this data on the page where the user is searching?

Reply:
This is so users don't need to know the slug of the project when using the search API. Specially from our extension, instead of having a query like

![Screenshot 2023-03-13 at 18-53-13 readthedocs-sphinx-search - Search as you type for Read the Docs — readthedocs-sphinx-search 0 2 0 documentation](https://user-images.githubusercontent.com/4975310/224857267-a1872378-5c20-456c-9f90-15f4a86bd8d9.png)

we can have a query without the slug

![Screenshot 2023-03-13 at 18-52-57 readthedocs-sphinx-search - Search as you type for Read the Docs — readthedocs-sphinx-search 0 2 0 documentation](https://user-images.githubusercontent.com/4975310/224857298-df0bff70-24ae-4db9-b446-605be142e1e3.png)

And for our extension is also easier to write the filters without having to know the slug of the project `project:@this` (but the setting could be dynamic to read the slug from the env vars we expose, so not really a blocker).

Slugs can be long, specially on .com.

If we don't want to go in this direction, we can:

- Just expose the slugs to users
- Hide the filters from the user, this is, just add the filter before hitting the API.